3차 플젝 첫번째 멘토링 리뷰

sobaman opened the pull request:
<!--
  템플릿은 아직 PR 작성이 익숙하지 않으신 분들을 위해서 제공하는 가이드입니다!
  리뷰어 또는 이 PR을 보게 될 다른 사람들이 이 PR을 보는데 참고할 수 있는 내용이 있다면 포함해서 작성해주시면 됩니다.
-->

## 📌 과제 설명 <!-- 어떤 걸 만들었는지 대략적으로 설명해주세요 -->

## 👩‍💻 요구 사항과 구현 내용 <!-- 기능을 Commit 별로 잘개 쪼개고, Commit 별로 설명해주세요 -->

## ✅ 피드백 반영사항  <!-- 지난 코드리뷰에서 고친 사항을 적어주세요. 재PR 시에만 사용해 주세요! (재PR 아닌 경우 삭제) -->

## ✅ PR 포인트 & 궁금한 점 <!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->

k-haechan committed:
refactor : MemberEntity에 refreshToken 필드 제거

Metronon committed:
feature: add redis caching DTO

Metronon committed:
refactor: modify LikesEntity

Metronon committed:
refactor: Comment Entity 반정규화

Metronon committed:
refactor: integrate exception

Metronon committed:
refactor: exception refactor

k-haechan committed:
feat: Add frontend authentication and navigation structure

k-haechan committed:
add: naver.svg 추가

Metronon committed:
refactor: toggle like logic

Metronon committed:
refactor: queryDSL config

Metronon committed:
refactor: separate OwnerCheck

Metronon committed:
refactor: separate Logics & Scheduler impl

ChoiHyunSan committed:
Feat: 피드 화면 구현 (API 호출 X)

Metronon committed:
refactor: like Controller & Service Test modify

Metronon committed:
test: add bulk query test

Metronon committed:
refactor: Bookmark, follow Test 예외 코드 수정

Metronon committed:
refactor: h2 frameOption disable

Metronon committed:
refactor: bean modify

Metronon committed:
refactor: test modify

sobaman committed:
feat: 검색 페이지 구현 (추후 API 호출 확인 필요)

sobaman committed:
refactor: notification 엔티티의 type 필드에 @Enumerated(EnumType.STRING) 추가

sobaman committed:
Merge pull request #56 from prgrms-be-devcourse/front/search-01

feat: 검색 페이지 구현 (추후 API 호출 확인 필요)

Metronon committed:
bugFix: 테스트 실패 일부 수정

Metronon committed:
feature: bookmarked list

Metronon committed:
test: bookmarked list test 작성

Metronon committed:
Merge branch 'main' of https://github.com/prgrms-be-devcourse/NBE3-4-2-Team2 into refactor/like

ChoiHyunSan committed:
Feat: 피드 컴포넌트 수정 및 백엔트 테스트 코드 수정

Commit:
feat: post 작성 frontend 구현

joonaeng committed:
feat: frontend create post 구현

Metronon committed:
refactor: modify Controller Test

k-haechan committed:
code : refreshToken field 제거

k-haechan committed:
refactor : username, password 로그인, 로그아웃 로직 구현 방식 변경 (security -> mvc)
chore : Swagger 웹 UI description 작업

k-haechan committed:
refactor : PasswordEncoder Bean 생성 위치 변경

k-haechan committed:
chore : rename CustomAuthenticationFailureHandler -> OAuth2FailureHandler
 CustomAuthenticationSuccessHandler -> OAuth2SuccessHandler

k-haechan committed:
feat : refreshToken을 생성하고 Set-Cookie설정까지 RefreshTokenService에 일임

k-haechan committed:
refactor : jwt claims에서 username을 제거하고 subject를 이용하도록 변경

k-haechan committed:
refactor : jwt 기반 로그인 구현 방식 변경

k-haechan committed:
refactor : 예외처리 위치 분리(401, 403)은 시큐리티에서 처리하도록 한다.

k-haechan committed:
refactor : 예외 양식 변경

k-haechan committed:
refactor : 토큰 생성 구현 변경

k-haechan committed:
feat : 로그인 기능 구현

k-haechan committed:
feat : 액세스 토큰 발급 API 생성

k-haechan committed:
feat : 프론트엔드 스키마 생성 로직 구현
- 테스트 시 빈 생성 문제 때문에 주석 처리

k-haechan committed:
frontend/auth commit

Metronon committed:
bugFix: Like관련 모든 테스트 통과하도록 수정

Metronon committed:
Merge pull request #54 from prgrms-be-devcourse/refactor/like

Refactor/like & feature bookmarked list

ChoiHyunSan committed:
Merge branch 'main' into frontend/feed-01

ChoiHyunSan committed:
Merge pull request #55 from prgrms-be-devcourse/frontend/feed-01

Feat: 피드 화면 구현 (API 호출 X)

k-haechan committed:
Merge branch 'main' into frontend/auth

k-haechan committed:
Merge pull request #58 from prgrms-be-devcourse/frontend/auth

Frontend/auth

Metronon committed:
bugFix: likes-> like & testData fix

ChoiHyunSan committed:
bugfix : Feed 테스트 오류 수정

ChoiHyunSan committed:
bugfix : JacksonConfig 파일 주석 처리

Metronon committed:
refactor: add join regex

k-haechan committed:
홈페이지 로그인 상태에 따라 분기 처리

joonaeng committed:
feat: 이미지 단건조회기능 추가

joonaeng committed:
feat: 이미지 조회기능 추가

Metronon committed:
Merge pull request #59 from prgrms-be-devcourse/frontend/auth-02

홈페이지 로그인 상태에 따라 분기 처리

Metronon committed:
feature: bookmark 페이지 (임시)

Metronon committed:
Merge branch 'main' of https://github.com/prgrms-be-devcourse/NBE3-4-2-Team2 into frontend/bookmark

sobaman committed:
feat: 알림 프론트 페이지 구현

sobaman committed:
Merge pull request #60 from prgrms-be-devcourse/frontend/notification-01

feat: 알림 프론트 페이지 구현

Metronon committed:
refactor: header&nav relative menu, dark mode add

sobaman committed:
refactor: 알림 도메인 리팩토링 및 테스트 케이스 추가

- 알림 도메인 api 앞에 "api-v1" 누락 부분 수정
- 알림 목록 불러오기 테스트 케이스 추가

sobaman committed:
Merge pull request #62 from prgrms-be-devcourse/refactor/notification-01

refactor: 알림 도메인 리팩토링 및 테스트 케이스 추가

joonaeng committed:
refactor: pos이미지 받아ì아오기 위해 포스트 파트 수정

sobaman committed:
refactor: 검색 쿼리 최적화

- 이중 이상 조인으로 데이터셋이 매우 커지는 점을 고려 쿼리를 두 개로 분리
- 서브쿼리를 어플리케이션 단에서 처리

Metronon committed:
refactor: dark mode 전역 관리로 변경

Metronon committed:
refactor: dark mode가 페이지 전체에 적용되도록 수정

Metronon committed:
refactor: Landing 페이지 테마 변경 적용되도록 수정

sobaman committed:
refactor: isSuccess 필드 -> success 수정

sobaman committed:
Merge pull request #65 from prgrms-be-devcourse/refactor/front/notification-01

refactor: isSuccess 필드 -> success 수정

Metronon committed:
refactor: schema에 따라 로직 수정 및 schema 최신화

Metronon committed:
refactor: dummyData likes -> like

Metronon commented on backend/src/main/java/com/example/backend/entity/LikeRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,169 @@
+
+		return query.executeUpdate();
+	}
+}
```
현재는 likeEntity로만 벌크 쿼리가 이뤄지고 있습니다.
하지만 저희 서비스에서 게시물(post), 댓글&대댓글(comment)에서 좋아요 갯수(likeCount)를 반정규화 해놓았기 때문에 이와 관련한 동기화 전략을 추가로 개발할 계획입니다.

1. 해당 likeEntity 벌크 쿼리에서 post, comment 접근을 통해 likeCount 업데이트 
2. 특정 주기마다 like 테이블에서 집계해 post, comment 의 likeCount 업데이트

성능 면에서 봤을때 2번이 비동기 방식으로 더 효율적일 것 같은데, 어떻게 생각하시는지 궁금합니다!

sobaman committed:
Merge pull request #63 from prgrms-be-devcourse/refactor/search-01

refactor: 검색 쿼리 최적화

sobaman commented on backend/src/main/java/com/example/backend/content/search/implement/SearchFinder.java:
```diff
@@ -43,26 +46,60 @@ public SearchPostCursorResponse findByKeyword(SearchType type, String keyword, L
+		Map<Long, SearchPostResponse> searchPostInitMap = new LinkedHashMap<>();
+		for (Long postId : postIds) {
+			searchPostInitMap.put(postId, SearchPostResponse.create(postId, null));
+		}
+
+		// 3. allImages 의 최대 크기는 (postIds * image 최대 개수)
+		// 성능 최적화) 1차 검색 결과에서 일치하는 postIds 가 없으면 실행하지 않음
+		if (!postIds.isEmpty()) {
+
+			List<ImageDataWithPostId> allImages = jpaQueryFactory
+				.select(Projections.constructor(
+					ImageDataWithPostId.class,
+					imageEntity.post.id,
+					imageEntity.imageUrl
+				))
+				.from(imageEntity)
+				.where(
+					imageEntity.post.id.in(postIds)
+					// 서브쿼리 삭제
+					// imageEntity.id.eq(
+					// 	JPAExpressions
+					// 		.select(imageEntity.id.min())
+					// 		.from(imageEntity)
+					// 		.where(imageEntity.post.id.eq(postEntity.id)))
+				)
+				.orderBy(imageEntity.post.id.desc(), imageEntity.id.asc()) // 이미지는 id 순서대로 저장된다고 가정
+				.limit(size + 1)
+				.fetch();
+
+			// 첫번째 이미지만을 searchPostInitMap 에 담기 위해 다음과 같이 처리
+			for (ImageDataWithPostId imageData : allImages) {
+				if (searchPostInitMap.get(imageData.postId()).imageUrl() == null) {
+					searchPostInitMap.put(imageData.postId(),
+						SearchPostResponse.create(imageData.postId(), imageData.imgUrl()));
+				}
+			}
+		}
+
+		List<SearchPostResponse> searchPostResponses = searchPostInitMap.values().stream().toList();
+
```
지난번 피드백 주신대로 이중 이상의 조인시 데이터셋이 매우 커지는 점을 고려해 쿼리를 분리하고, 서브쿼리를 어플리케이션 레벨에서 처리하는 식으로 변경해봤습니다. 그러다보니 전체를 allImages 로 다 들고 와서 첫번째 이미지만 넣는 식으로 처리가 됐는데요 , 현재는 imgUrl 의 최대 개수가 많아봤자 10개인 상황이라 한번에 들고 와서 처리하는게 크게 부담이 되지 않는다고 판단하였습니다. 실제로도 서브쿼리의 성능 문제 개선을 위해 이런점들을 고려하여 어플리케이션 레벨에서 처리하는 지 궁금합니다

Metronon commented on backend/src/main/java/com/example/backend/social/reaction/like/service/LikeService.java:
```diff
@@ -0,0 +1,102 @@
+
+		return LikeConverter.toLikeResponse(likeInfo, likeCount);
+	}
+}
```
좋아요 서비스 부분에서 사용되는 컴포넌트 부분이 방대해 각각의 클래스로 나눠서 사용하고 있습니다.
유지보수 측면에서 현재 작업한 코드와 같이 나눈게 적당할까요?

Metronon committed:
Merge pull request #64 from prgrms-be-devcourse/frontend/bookmark

Frontend/bookmark & etc

joonaeng committed:
feat: 게시물 등록하기 api 연동중

joonaeng committed:
resolved merge conflict

joonaeng committed:
현재 작업중 저장

joonaeng committed:
중간 저장

sobaman committed:
feat: sse 알림 부분 프론트 페이지 구현

k-haechan reviewed (commented):
401예외와 403예외는 스프링 시큐리티를 사용하는 거면 되도록 스프링 시큐리티에서 처리하는게 좋은건가요??
개발자의 구현 방식에 따라 다르겠지만 
1. SpringSecurity에서 예외를 Spring MVC로 던져서 GlobalExceptionHandler에서 한번에 처리하도록 한다.
2. 401과 403은 SpringSecurity에서 예외를 던진다.
이 두가지 관점 중 유지보수와 코드 관리에 용이한 관점이 무엇인지 궁금합니다!

joonaeng committed:
중간저장2

ChoiHyunSan committed:
feat: 프론트앤드 피드 API 요청 적용

ChoiHyunSan committed:
Merge pull request #67 from prgrms-be-devcourse/frontend/feed-03

feat: 프론트앤드 피드 API 요청 적용

joonaeng committed:
pull main

sobaman committed:
Merge branch 'main' into front/notification-02

joonaeng committed:
feat: post 기능 확인 (이미지, memberId 제외)

ChoiHyunSan committed:
feat: Feed 컴포넌트에서 좋아요 & 북마크 API를 요청하도록 handle 메서드 수정

ChoiHyunSan committed:
bugfix: 처음 요청하는 Feed 요청이 제대로 이뤄지지 않는 버그 수정

sobaman committed:
feat: sse 연결 프론트 부분 구현

- 컴포넌트, 훅 구현
- 테스트를 위한 데이터 생성
- 버튼을 누르면 sse 연결이 재설정 되지 않게끔 구현

sobaman committed:
feat: provider 적용 및 테스트 버튼 생성

- 레이아웃에 sse 알림 Provider 적용
- 로그인 후 메인 페이지에 알림 테스트 버튼 구현

sobaman committed:
Merge pull request #68 from prgrms-be-devcourse/front/notification-02

Front/notification 02

joonaeng committed:
ㄹfeat: 이미지 업로드 API 연동, DB에 파일명만 저장

joonaeng committed:
feat: 게시물 만들기 API 연동 완료

joonaeng committed:
remove .idea

joonaeng committed:
remove .idea

joonaeng committed:
Remove .idea and testImages

joonaeng committed:
Remove formatter

Metronon committed:
refactor: follow 로직 단순화

Metronon committed:
refactor: String타입의 상대 이름 가져오도록 변경

Metronon committed:
refactor: service 로직 수정

Metronon committed:
refactor: 맞팔로우 확인 로직 수정

sobaman committed:
refactor: 백엔드 서버 포트 누락

- 백엔드 포트 주소 추가

sobaman committed:
refactor: 불변 리스트에서 removeLast() 가 적용되는 부분 수정

- hasNext 위해 size+1 만큼 가져와 다음 페이지가 존재하면 마지막 원소 제거하는 과정에서 불변리스트라 remove 가 안되는 부분을 subList() 로 반환해 수정

sobaman committed:
Merge pull request #70 from prgrms-be-devcourse/refactor/front/search-01

refactor: 백엔드 서버 포트 누락

sobaman committed:
Merge pull request #71 from prgrms-be-devcourse/refactor/search/searchfinder

refactor: 불변 리스트에서 removeLast() 가 적용되는 부분 수정

Metronon committed:
test: 테스트 코드 수정

Metronon committed:
refactor: 사용하지 않는 파일 제거, feed test init data 수정

ChoiHyunSan committed:
refactor: Feed에 추가적으로 필요한 정보를 Dto에 추가하고 연관된 클래스 수정

ChoiHyunSan committed:
refactor: Feed 상세(단건) 페이지를 모달 방식으로 변경하여 나타내도록 수정

ChoiHyunSan committed:
delete: 필요 없어진 FeedDetail.tsx 삭제

Metronon committed:
bugFix: 오류 부분 수정 (JPA 연관관계)

ChoiHyunSan committed:
refactor: 주석 수정 및 불필요한 파일 삭제

ChoiHyunSan committed:
Merge pull request #69 from prgrms-be-devcourse/frontend/post-01

Frontend/post 01

Metronon committed:
Merge branch 'main' into refactor/follow

Metronon committed:
Merge pull request #72 from prgrms-be-devcourse/refactor/follow

Refactor/follow

ChoiHyunSan committed:
Merge branch 'main' into frontend/feed-03

ChoiHyunSan committed:
Merge pull request #73 from prgrms-be-devcourse/frontend/feed-03

Frontend/feed 03

ChoiHyunSan committed:
refactor: 피드 좋아요 기능 추가 및 페이징 처리 수정

ChoiHyunSan committed:
Merge pull request #74 from prgrms-be-devcourse/frontend/feed-04

refactor: 피드 좋아요 기능 추가 및 페이징 처리 수정

Metronon committed:
bugFix: equal 오류부분 수정

ChoiHyunSan committed:
bugfix: 피드 모달파일 추가

ChoiHyunSan committed:
Merge pull request #75 from prgrms-be-devcourse/frontend/feed-04

bugfix: 피드 모달파일 추가

Metronon committed:
refactor: test 통과하도록 로직 수정

Metronon committed:
Merge pull request #76 from prgrms-be-devcourse/refactor/follow

Refactor/follow

Metronon committed:
.gitignore 수정

Metronon committed:
fix git cached

Metronon committed:
이미지 경로 허용작업

k-haechan committed:
bugfix : redis의 key는 not null

Metronon committed:
front feature: imageUrl 확인 util화 및 적용

ChoiHyunSan committed:
Feat: 피드 상세 모달창에도 이미지가 정상적으로 보이도록 수정

ChoiHyunSan committed:
Feat: 검색 창 이미지도 정상적으로 나오도록 수정

Metronon committed:
feature: jwt parse util 추가

Metronon committed:
refactor: post 페이지 수정(다크모드, 스크롤 문제, 유저id 주입)

ChoiHyunSan committed:
feat: 좋아요/ 북마크 기능이 메인피드와 모달 창에서 정상적으로 연동되게끔 수정

Metronon committed:
refactor: post 페이지 제목 좌우 정렬 수정

ChoiHyunSan committed:
refactor: 모달 창 테마 변경

joonaeng committed:
Merge pull request #78 from prgrms-be-devcourse/frontend/refactor

Frontend/refactor

sobaman committed:
refactor: comment 이벤트 관련 수정

- 프론트에서 댓글 알림을 누르면 해당 post 의 댓글로 이동해야하는데, postId 를 알 수가 없어서 msg 에 postid 를 같이 담아 보내는 식으로 수정

Metronon committed:
Merge pull request #77 from prgrms-be-devcourse/feature/security-bugfix

bugfix : redis의 key는 not null

ChoiHyunSan committed:
Merge pull request #79 from prgrms-be-devcourse/frontend/feed-04

Frontend/feed 04

Metronon committed:
refactor: 피드 다크모드 적용

Metronon committed:
Merge pull request #80 from prgrms-be-devcourse/frontend/refactor

refactor: 피드 다크모드 적용

joonaeng committed:
refactor: comment 기능 개선

joonaeng committed:
refactor: comment 기능 개선

ChoiHyunSan committed:
refactor: BaseEntity 코틀린 접근을 위해 Getter 메서드 Lombok 사용대신 직접 선언

ChoiHyunSan committed:
refactor: Feed 도메인 코틀린화 작업

ChoiHyunSan committed:
refactor: Feed 도메인 코틀린화 작업

ChoiHyunSan committed:
bugfix: AsyncConfig 클래스를 추가하여 @Async 에노테이션이 비동기적으로 정상 처리하도록 수정

k-haechan committed:
feat : AccessToken을 통해 id 추출

k-haechan committed:
Merge pull request #82 from prgrms-be-devcourse/frontend/hot-fix

feat : AccessToken을 통해 id 추출

sobaman committed:
Merge pull request #83 from prgrms-be-devcourse/refactor/notification/comment

refactor: comment 이벤트 관련 수정

ChoiHyunSan committed:
bugfix: FeedService 메인 피드 기능을 제외한 다른 메서드가 정상적으로 호출되지 않는 오류 수정

ChoiHyunSan committed:
feat: 검색 창에서 단건 모달 창이 띄워지도록 수정

ChoiHyunSan committed:
feat: 코틀린 세팅 추가

ChoiHyunSan committed:
feat: 북마크 창에서 단건 모달 창이 띄워지도록 수정

k-haechan committed:
feat : profile페이지 ver.1

ChoiHyunSan committed:
refactor: 피드 테스트 코드 수정

ChoiHyunSan committed:
Merge pull request #84 from prgrms-be-devcourse/frontend/feed-06

Frontend/feed 06

joonaeng committed:
Merge branch 'main' into frontend/comment-01

joonaeng committed:
댓글 목록 API 연동

joonaeng committed:
댓글 목록 API 연동

Metronon committed:
feature: like local save util

Metronon committed:
refactor: feed는 로컬의 좋아요, 북마크 값을 먼저 확인한다

Metronon committed:
feature:likeCount 동기화 쿼리

Metronon committed:
refactor: 데이터가 방대해짐에 따라 서브쿼리 채용으로 부하 최적화

Metronon committed:
Merge pull request #85 from prgrms-be-devcourse/frontend/auth-04

feat : profile페이지 ver.1

Metronon committed:
Merge pull request #86 from prgrms-be-devcourse/refactor/like-sync

Refactor/like sync

joonaeng committed:
refactor: 댓글 구조 개선

joonaeng committed:
refactor: 댓글 구조 개선

joonaeng committed:
feat: 대댓글 기능 구현 & 댓글 트리 구조 구현 중

joonaeng committed:
feat: 대댓글 기능 구현 & 댓글 트리 구조 구현 중

joonaeng committed:
feat: 댓글 트리구조 구현 완료 & CommentEntity의 ParentId -> ParentNum으로 전부 수정(프론트와 충돌 이벤트)

joonaeng committed:
feat: 댓글 트리구조 구현 완료 & CommentEntity의 ParentId -> ParentNum으로 전부 수정(프론트와 충돌 이벤트)

joonaeng committed:
feat: 댓글 트리 구조 구현 완료, 페이징 구현 X(차후에 할 예정, 댓글 50개 불러오기로 임시 수정) & DTO parentNum nullable, 스키마 memberId nullable

joonaeng committed:
feat: 댓글 트리 구조 구현 완료, 페이징 구현 X(차후에 할 예정, 댓글 50개 불러오기로 임시 수정) & DTO parentNum nullable, 스키마 memberId nullable

Metronon committed:
refactor: .gitignore 수정, Event Test bugFix

joonaeng committed:
feat: 댓글 수정/ 삭제 기능 구현중

joonaeng committed:
Merge branch 'main' into frontend/comment-01

joonaeng committed:
merge: merge complete

k-haechan committed:
feat : API 엔드포인트 변경 member/{id} -> member/{username}

k-haechan committed:
Merge pull request #87 from prgrms-be-devcourse/feature/fix

feat : API 엔드포인트 변경 member/{id} -> member/{username}

ChoiHyunSan committed:
bugfix: 병합 후 모달 창이 정상적으로 열리지 않는 오류 수정

ChoiHyunSan committed:
refactor: 프로필 게시물을 누르면 게시물 모달창이 열리도록 수정

ChoiHyunSan committed:
bugfix: 모달창 상호작용이 정상적으로 이뤄지지 않는 현상 수정

ChoiHyunSan committed:
Merge pull request #88 from prgrms-be-devcourse/frontend/feed-07

Frontend/feed 07

Commit:
rebase main

Commit:
Merge remote-tracking branch 'origin/frontend/comment-01' into frontend/comment-01

# Conflicts:
#	frontend/src/components/feed/Comment.tsx
#	frontend/src/components/feed/useComments.ts

sobaman committed:
refactor: 알림 생성과 전송 로직 분리

- 알림 생성과 전송 로직 분리
- 분리에 따른 각 이벤트 리스너 수정

sobaman committed:
refactor: 각 브라우저 당 하나의 sse connection 허용

- 브라우저 이름을 파라미터로 전달받게하여 동일 유저의 브라우저를 구분

sobaman committed:
refactor: 프론트 수정

sobaman committed:
Merge pull request #89 from prgrms-be-devcourse/refactor/sse

Refactor/sse

Metronon committed:
refactor: follow service migration

k-haechan committed:
bugfix : 로그아웃 버그 fix

k-haechan committed:
Merge pull request #90 from prgrms-be-devcourse/feature/fix

bugfix : 로그아웃 버그 fix

Metronon committed:
feature: follow 엔드포인트 추가

Metronon committed:
refactor: bookmark service 코틀린화

Metronon committed:
LikeStateInfo DTO 분리

Metronon committed:
refactor: LikeService 코틀린 마이그레이션

Metronon committed:
Merge pull request #91 from prgrms-be-devcourse/kotlin/social

Kotlin/social

sobaman committed:
refactor: sse 401 문제로 인해 인증/인가에서 제외

- userId 는 파라미터로 넘겨받는 것으로 변경
- 시큐리티 설정에서 permitAll 로 변경

sobaman committed:
refactor: 프론트도 userId 를 파라미터로 넘기도록 설정

sobaman committed:
Merge pull request #92 from prgrms-be-devcourse/refactor/sse-02

Refactor: sse 인증/인가에서 제외

sobaman committed:
feat: hashtag 서비스 코틀린으로 변경

- 기존 자바 버전 서비스 삭제 및 코틀린으로 변경

Commit:
merge complete

sobaman committed:
feat: 검색 서비스 코틀린으로 변경

- 기존 자바 기반 검색 서비스를 코틀린으로 변경

sobaman committed:
Merge pull request #93 from prgrms-be-devcourse/kotlin/hashtag-01

feat: hashtag 서비스 코틀린으로 변경

sobaman committed:
Merge pull request #94 from prgrms-be-devcourse/kotlin/search

feat: 검색 서비스 코틀린으로 변경

Metronon committed:
refactor: feed 기본 이미지 설정

Metronon committed:
Merge branch 'main' of https://github.com/prgrms-be-devcourse/NBE3-4-2-Team2

k-haechan committed:
feat : 프로필 페이지 구현

Metronon committed:
refactor: fetchComments

k-haechan committed:
Merge pull request #95 from prgrms-be-devcourse/frontend/auth-05

프로필 페이지 추가 구현

sobaman committed:
feat: 알림 서비스 코틀린화 및 엔티티에 get 메서드 추가

- 알림 서비스 코틀린화
- 엔티티에 get 메서드 추가

sobaman committed:
Merge pull request #96 from prgrms-be-devcourse/kotlin/notification

feat: 알림 서비스 코틀린화 및 엔티티에 get 메서드 추가

sobaman committed:
refactor: searchfinder 의존성 적절히 주입되게 수정

sobaman committed:
Merge pull request #97 from prgrms-be-devcourse/kotlin/refactor/search-01

refactor: searchfinder 의존성 적절히 주입되게 수정

sobaman committed:
refactor: hashtag 코틀린화 수정

- !!삭제
- isNotEmpty() 로 변경
- 롬복 의존성 삭제

sobaman committed:
Merge pull request #98 from prgrms-be-devcourse/kotlin/refactor/hashtag-01

refactor: hashtag 코틀린화 수정

k-haechan committed:
style : 코드 구조 변경

k-haechan committed:
Merge pull request #99 from prgrms-be-devcourse/feature/fix

style : 코드 구조 변경

sobaman committed:
refactor: 페이지 목록 메서드에 트랜잭션 누락 수정

sobaman committed:
Merge pull request #100 from prgrms-be-devcourse/kotlin/refactor/getNotiPage

refactor: 페이지 목록 메서드에 트랜잭션 누락 수정

Commit:
feat: 댓글 대댓글 좋아요 기능 구현

Commit:
refactor: 좋아요 로컬 저장 변수 변경

Commit:
delete schema.d.ts

Commit:
add schema.d.ts

joonaeng committed:
Merge pull request #101 from prgrms-be-devcourse/frontend/comment-01

Frontend/comment 01

k-haechan committed:
refactor : kotlin 변경

k-haechan committed:
refactor : SecurityConfig.java -> SecurityConfig.kt

k-haechan committed:
chore : 페이지 이름, favicon 설정

k-haechan committed:
Merge pull request #102 from prgrms-be-devcourse/feature/fix

코틀린 전환, 웹 페이지 이름 등록

ChoiHyunSan committed:
bugfix: 피드 프로필 이미지를 누르면 프로필 이미지로 이동되도록 수정

ChoiHyunSan committed:
Merge pull request #103 from prgrms-be-devcourse/frontend/feed-10

bugfix: 피드 프로필 이미지를 누르면 프로필 이미지로 이동되도록 수정

Metronon committed:
refactor: dark mode

Commit:
Rename .java to .kt

Commit:
refactor: Post, Image, Comment Service 코틀린 변환

joonaeng committed:
Merge pull request #104 from prgrms-be-devcourse/frontend/comment-01

Frontend/comment 01
